phoenix-core/src/main/java/org/apache/phoenix/exception/DataExceedsCapacityException.java

```diff
@@ -31,11 +31,26 @@ public DataExceedsCapacityException(String message) {
                 SQLExceptionCode.DATA_EXCEEDS_MAX_CAPACITY).setMessage(message).build().buildException());
     }
     
+    @Deprecated
     public DataExceedsCapacityException(PDataType type, Integer precision, Integer scale, String columnName, ImmutableBytesWritable value) {
         super(new SQLExceptionInfo.Builder(SQLExceptionCode.DATA_EXCEEDS_MAX_CAPACITY)
             .setMessage((columnName == null ? "" : columnName + " ") + getTypeDisplayString(type, precision, scale, value))
             .build().buildException());
     }
+
```

Comment:
You could instead add the new constructor as:

```
    public DataExceedsCapacityException(PDataType type, Integer precision, Integer scale, String columnName) {
        super(new SQLExceptionInfo.Builder(SQLExceptionCode.DATA_EXCEEDS_MAX_CAPACITY)
                .setMessage((columnName == null ? "" : columnName + " ") + getTypeDisplayString(type, precision, scale, null))
                .build().buildException());
    }
```
And, for example, throw the exception as:
```
    throw new DataExceedsCapacityException(column.getDataType(), column.getMaxLength(),
        column.getScale(), column.getName().getString());
```